assets/app/view/game/part/city.rb

```diff
@@ -241,20 +323,33 @@ def render_revenue
 
           rotation = 0
 
-          case @num_cities
-          when 1
-
+          if @num_cts == 1
             rotation = angle_for_layout
 
             regions = if layout == :flat
                         @city.slots == 1 ? [9, 16] : [11, 18]
                       else
                         @city.slots == 1 ? [8, 9] : [3, 4]
                       end
-          when 2
-            if @edge
-              regions, negative_displacement = OO_REVENUE_REGIONS[@edge]
-              displacement *= -1 if negative_displacement
+          elsif @edge && @city.slots == 1
+            regions, rotation = find_revenue_location(@edge)
+          elsif @edge
+            regions, negative_displacement = OO_REVENUE_REGIONS[@edge]
+            displacement *= -1 if negative_displacement
+          else
+            # pick an edge where there isn't another stop
+            other_stops = (@tile.cities + @tile.towns).reject { |s| s == @city }
```

Comment:
this whole else statement seems inefficient

edges = SOME_CONSTANT - (@tiles.cities + @tile.towns).flat_map do |stop|
    next [] if stop == @city || !(edge = @tile.preferred_city_town_edges[stop])
    [edge, (edge - 1) % 6]
end

revenue_edge = edges[0] || 0
rotation = 60 * revenue_edge + 120
regions = CENTER_REVENUE_REGIONS[revenue_edge]

Reply:
OK, I'll recode.